Unhandled interface conversion error

**Versions**

HC's provider: 0.11.0

**Steps to reproduce**

I'm trying to deploy some boards to monitor the performance of our RDS databases, in several AWS environments. For most of our environments, the boards deploy exactly like we want, but for one in particular I'm getting a panic:

```
╷
│ Error: Request cancelled
│
│   with module.microservice_group.module.this["privacy_service"].module.rds_board[0].module.this.honeycombio_board.this,
│   on ../platform-component-hc-board/generic-board/main.tf line 64, in resource "honeycombio_board" "this":
│   64: resource "honeycombio_board" "this" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-honeycombio_v0.11.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 79 [running]:
github.com/honeycombio/terraform-provider-honeycombio/honeycombio.expandBoardQueryGraphSettings({0x170b0e0?, 0xc00061b320?})
	github.com/honeycombio/terraform-provider-honeycombio/honeycombio/resource_board.go:254 +0x29f
github.com/honeycombio/terraform-provider-honeycombio/honeycombio.expandBoard(0xc0003dac00)
	github.com/honeycombio/terraform-provider-honeycombio/honeycombio/resource_board.go:215 +0x4e5
github.com/honeycombio/terraform-provider-honeycombio/honeycombio.resourceBoardUpdate({0x1944a08, 0xc0005a8000}, 0x0?, {0x173d1a0?, 0xc00022c100?})
	github.com/honeycombio/terraform-provider-honeycombio/honeycombio/resource_board.go:181 +0x5d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0002270a0, {0x1944a40, 0xc0004b9140}, 0xd?, {0x173d1a0, 0xc00022c100})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0002270a0, {0x1944a40, 0xc0004b9140}, 0xc000598340, 0xc0003dbe80, {0x173d1a0, 0xc00022c100})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:847 +0x82c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000dd40, {0x1944a40?, 0xc0004b9020?}, 0xc00043d900)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00024c0a0, {0x1944a40?, 0xc0004b84b0?}, 0xc0003ea230)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x17f5c40?, 0xc00024c0a0}, {0x1944a40, 0xc0004b84b0}, 0xc0003ea000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000003c0, {0x1947920, 0xc0004141a0}, 0xc000672900, 0xc0002501e0, 0x1dc6b20, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb0b
google.golang.org/grpc.(*Server).handleStream(0xc0000003c0, {0x1947920, 0xc0004141a0}, 0xc000672900, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

Error: The terraform-provider-honeycombio_v0.11.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

We pinned the version of the provider to 0.11.0 recently because the latest version prevents us from deploying, but for another reason (unprocessable entities related to alerts, I believe).

**Additional context**

I believe the error comes from this line:

https://github.com/honeycombio/terraform-provider-honeycombio/blob/84525914331cf6cb9d57d0b9682dd57431d4529b/honeycombio/resource_board.go#L254

It looks like `raw[0]` is nil.


The boards that we're trying to deploy:

```yaml
resource "honeycombio_query" "this" {
  for_each = var.queries

  depends_on = [
    data.honeycombio_query_specification.this
  ]

  dataset    = coalesce(each.value.meta.dataset, var.default_dataset)
  query_json = data.honeycombio_query_specification.this[each.key].json
}

resource "honeycombio_query_annotation" "this" {
  for_each = var.queries

  depends_on = [
    honeycombio_query.this
  ]

  dataset  = honeycombio_query.this[each.key].dataset
  query_id = honeycombio_query.this[each.key].id
  name     = each.value.meta.annotation
}

resource "honeycombio_board" "this" {
  depends_on = [
    honeycombio_query.this,
    honeycombio_query_annotation.this
  ]

  name  = var.name
  style = "visual"

  description = var.description

  dynamic "query" {
    for_each = var.queries

    content {
      dataset             = honeycombio_query.this[query.key].dataset
      caption             = query.value.meta.caption
      query_id            = honeycombio_query.this[query.key].id
      query_annotation_id = honeycombio_query_annotation.this[query.key].id
      query_style         = query.value.meta.style

      graph_settings {
        hide_markers        = try(query.value.graph_settings.hide_markers, null)
        log_scale           = try(query.value.graph_settings.log_scale, null)
        omit_missing_values = try(query.value.graph_settings.omit_missing_values, null)
        stacked_graphs      = try(query.value.graph_settings.stacked_graphs, null)
        utc_xaxis           = try(query.value.graph_settings.utc_xaxis, null)
      }
    }
  }
}
```

and `_variables.tf`:

```yaml
variable "queries" {
  type = map(object({
    meta = object({
      annotation = string
      caption    = optional(string)
      dataset    = optional(string) # defaults to var.default_dataset
      style      = optional(string) # defaults to graph
    })
    # defaults to COUNT
    calculations = optional(list(object({
      op     = string
      column = optional(string)
    })))
    filters = list(object({
      column = string
      op     = string
      value  = optional(any)
    }))
    breakdowns = optional(list(string))
    orders = optional(list(object({
      op     = optional(string)
      column = optional(string)
      order  = optional(string)
    })))
    limit = optional(number)
    graph_settings = optional(object({
      hide_markers        = optional(bool)
      log_scale           = optional(bool)
      omit_missing_values = optional(bool)
      stacked_graphs      = optional(bool)
      utc_xaxis           = optional(bool)
    }))
    granularity = optional(number)
  }))
  description = "Object that defines queries to be added to the board"
}
```

--- a/honeycombio/resource_board.go
+++ b/honeycombio/resource_board.go
@@ -251,6 +251,9 @@ func expandBoardQueryGraphSettings(gs interface{}) (honeycombio.BoardGraphSettin
 	if len(raw) > 1 {
 		return graphSettings, errors.New("got more than one set of graph settings?")
 	}
+	if _, ok := raw[0].(map[string]interface{}); !ok {
+		return graphSettings, nil
+	}
 	s := raw[0].(map[string]interface{})
 
 	if v, ok := s["log_scale"].(bool); ok && v {

--- a/honeycombio/resource_board.go
+++ b/honeycombio/resource_board.go
@@ -251,10 +251,10 @@ func expandBoardQueryGraphSettings(gs interface{}) (honeycombio.BoardGraphSettin
 	if len(raw) > 1 {
 		return graphSettings, errors.New("got more than one set of graph settings?")
 	}
-	if _, ok := raw[0].(map[string]interface{}); !ok {
+	s, ok := raw[0].(map[string]interface{})
+	if !ok {
 		return graphSettings, nil
 	}
-	s := raw[0].(map[string]interface{})
 
 	if v, ok := s["log_scale"].(bool); ok && v {
 		graphSettings.UseLogScale = true

